include/ap/argument_parser.hpp

```diff
@@ -1474,77 +1507,36 @@ class argument_parser {
-            return this->discriminator == other.discriminator and this->value == other.value;
-        }
-
-        type_discriminator discriminator;
-        std::string value;
-    };
-
-    using cmd_argument_list = std::vector<cmd_argument>;
-    using cmd_argument_list_iterator = typename cmd_argument_list::const_iterator;
-
-    using argument_ptr_type = std::unique_ptr<argument::detail::argument_interface>;
-    using argument_opt_type = std::optional<std::reference_wrapper<argument::detail::argument_interface>>;
-    using argument_list_type = std::vector<argument_ptr_type>;
-    using argument_list_iterator_type = typename argument_list_type::iterator;
-    using argument_list_const_iterator_type = typename argument_list_type::const_iterator;
-    using argument_predicate_type = std::function<bool(const argument_ptr_type&)>;
-
     /**
-     * @brief Function to create a predicate for finding arguments by name.
+     * @brief Function to create a predicate for finding arguments by thename.
      * @param name The name of the argument.
      * @return Argument predicate based on the provided name.
      */
-    [[nodiscard]] inline argument_predicate_type _name_eq_predicate(const std::string_view& name) const {
+    [[nodiscard]] argument_predicate_type _name_eq_predicate(const std::string_view& name) const {
         return [&name](const argument_ptr_type& arg) { return name == arg->name(); };
     }
 
     /**
-     * @brief Function to create a predicate for finding arguments by name and short name.
-     * @param name The name of the argument.
-     * @param short_name The short name of the argument.
-     * @return Argument predicate based on the provided name and short name.
+     * @brief Function to create a predicate for finding arguments by the name.
+     * @param primary_name The name of the argument.
+     * @param secondary_name The secondary name of the argument.
+     * @return Argument predicate based on the provided name and secondary name.
      */
-    [[nodiscard]] inline argument_predicate_type _name_eq_predicate(
-        const std::string_view& name, const std::string_view& short_name
+    [[nodiscard]] argument_predicate_type _name_eq_predicate(
+        const std::string_view& primary_name, const std::string_view& secondary_name
     ) const {
-        return [&name, &short_name](const argument_ptr_type& arg) {
-            return name == arg->name() or short_name == arg->name();
+        return [&primary_name, &secondary_name](const argument_ptr_type& arg) {
+            return primary_name == arg->name() or secondary_name == arg->name();
         };
     }
 
     /**
      * @brief Check if an argument name is already used.
      * @param name The name of the argument.
-     * @return True if the argument name is already used, false otherwise.
+     * @return True if the argument name is already used.
      */
     [[nodiscard]] bool _is_arg_name_used(const std::string_view& name) const {
```

Comment:
Replace the _is_arg_name_used functions like described above for the _name_eq_predicate